boto3/session.py

```diff
@@ -350,3 +351,7 @@ def _register_default_handlers(self):
             'creating-resource-class.ec2.ServiceResource',
             boto3.utils.lazy_call(
                 'boto3.ec2.createtags.inject_create_tags'))
+
+        delete_tags = create_delete_tags_action(self.events)
+        self._session.register(
```

Comment:
I like james's idea of being able to inject kwargs as a parameter to lazy_call. Definately worth adding to keep lazy registering the handlers.